### DISCUSSION TITLE
Cutting and engraving in once

### DISCUSSION THREAD

**Discussion by Pkmngtr8** [arkypita/LaserGRBL#1244] [Q&A]
Is it possible in this program somehow cut and engrave in once?
Some lines will be multiple times cut, but the other just once or twice engraved.
What is the best way make something similar? https://support.glowforge.com/hc/en-us/articles/360033633394-3-Make-a-Gift-Tag-from-Scratch

**Comment by arkypita** [comment]
LaserGRBL is a simple gcode generate and stream, without advanced CAD/CAM features.
Nothing prevents these functions from being implemented in the future, but at the moment it is not possible.

Ps. you can always create your work in several parts and load them with different settings via the "file - append" menu

**Comment by vuchl** [comment]
@Pkmngtr8 I thought about this also. Maybe this can be solved for the bounding box or a circular shape via the customizable buttons.

The Frame buttons code could be copied and adjusted to set the laser intesity to 100% and then iterate a few times over it. A circle around the complete engraving could be done via the center function and the some GCODE for cricles plus an intensity of 100%.

**Reply by bman2000** [reply]
It would be nice if at the end of an engrave/ cut and option to reopen the settings, not file, so that you can vector cut the image outline. Reloading the image is not good enough if the size has been altered.

**Reply by vuchl** [reply]
Yes this also sounds like a good idea! Keeping track of the settings is a bit of a nuisance. This would require an adjustment in the code of LaserGRBL. 

@arkypita is the outline somehow exposable to the action buttons? I think propably not since it is a different "rasterization".

**Reply by bman2000** [reply]
Could you create an internal file, containing all the settings, and then reload it at the end of the process?

**Comment by arkypita** [comment]
I thought I had already explained it in the previous message.
LaserGRBL is not an image drawing and processing program, nor is it a engraving project manager.

All commands on the left side of editing (brightness, contrast, tool) are stored and restored for the next file (be it the same or a new one)

The few editing functions it provides (crop, flip, rotate) are instead destructive, in the sense that they transform the image into memory, and then when the gcode is generated the previous work is no longer available. It is not so easy to do otherwise. I've tried, even recently, but it's not easy. Furthermore, these operations are better if they are done in a dedicated graphics program. And they are certainly more complete.

What you are asking for is a project manager. A system that allows you to build a work made up of several parts/layer so that for each object/part/layer you can set up a different process, and which can be modified in its parts.

It's an interesting thing, it's already in my development roadmap.
Maybe one day there will be, but not now.

**Comment by arkypita** [comment]
What I can do, as an intermediate step, is to try to keep the result of the elaborations persistent (crop / flip etc) and invalidate them if you change files, but as I have already explained it is a bit complex.

**Comment by bman2000** [comment]
I was hoping that we could save the file location and size so that we can
perform another operation, like vectorized, on the already processed image.

I appreciate all your work. It's a great program.

Thank you

Bernie
